framework/src/bcs/IntegratedBCBase.C

```diff
@@ -61,3 +72,27 @@ IntegratedBCBase::prepareShapes(const unsigned int var_num)
 {
   _subproblem.prepareFaceShapes(var_num, _tid);
 }
+
+bool
+IntegratedBCBase::shouldApply() const
+{
+#ifndef NDEBUG
```

Comment:
```suggestion
#ifdef DEBUG
```
We typically add O(1) assertion like checks for devel and restrict checks that depend on N (N in this case the number of subdomains) to debug. Admittedly it's pretty rare to have a lot of subdomains

Reply:
I think I have it right right now
https://github.com/microsoft/qsharp-runtime/issues/600

NDEBUG should be defined. DEBUG may be defined

Reply:
oh nevermind I see what you mean

Reply:
I think searching in a set is log(n) so should be fine to have in devel

Reply:
same I use devel mostly. I can go either way. if you find it slow sure let s change it to debug

Reply:
probably not worth the time we're spending over it

Reply:
sure. I tried `#ifdef DEBUG` and I am not seeing it trigger in devel

also this does not work
```
    method = 'dbg|devel'
```

I ll leave it as is

Reply:
ooooh